### PR TITLE
Added Manual Dark Mode toggle

### PR DESCRIPTION
**Pull request by justdvnsh**
Solved issue #73. Now dark mode can be toggled on and off through home screen !

Video sample -> 
https://drive.google.com/file/d/1-gX9wL-7GrbeAjrzWsmteUBT23DiQc4c/view?usp=sharing

### CODE REVIEW

**Comment by mukul500** [comment]
Don't put icon on top.

**Comment by justdvnsh** [comment]
yeah , but then how would we toggle into dark mode ?

File: app/src/main/java/net/xblacky/animexstream/ui/main/player/VideoPlayerFragment.kt [hunk 1]
```diff
@@ -338,8 +348,46 @@ class VideoPlayerFragment : Fragment(), View.OnClickListener, Player.EventListen
...
+            setPositiveButton("OK") {dialog, _ ->
+                setPlaybackSpeed(speeds[selectedSpeed])
+                dialog.dismiss()
+            }
+            setNegativeButton("Cancel") {dialog, _ ->
+                dialog.dismiss()
+            }
+        }
+        val dialog = builder.create()
+        dialog.show()
```
Comment (mukul500) [inline app/src/main/java/net/xblacky/animexstream/ui/main/player/VideoPlayerFragment.kt:390]: Why is this code repeated again in this commit

File: app/.gitignore [hunk 2]
```diff
@@ -1 +1,2 @@
 /build
+google-services.json
```
Comment (mukul500) [inline app/.gitignore:2]: Why there are 2 separate .gitingores

File: app/src/main/java/net/xblacky/animexstream/ui/main/player/VideoPlayerFragment.kt (hunk 1 above)
Comment (justdvnsh) [reply app/src/main/java/net/xblacky/animexstream/ui/main/player/VideoPlayerFragment.kt:390]: yeah. even I was wondering the same. will look after it tomorrow. Also add the volume and brightness control functionality in tomorrow's pull request

**Comment by GoluLiliput** [comment]
Maybe icon can be made dynamic. Like the sun icon when the app is dark mode and moon icon even the app is in light mode..

File: app/src/main/res/layout/fragment_home.xml [hunk 3]
```diff
@@ -48,6 +48,20 @@
...
+        android:padding="10dp"
+        android:layout_marginTop="5dp"
+        android:layout_marginEnd="5dp"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintEnd_toStartOf="@id/favorite"
+        android:src="@drawable/ic_baseline_brightness_3_24"
+        android:background="?selectableItemBackgroundBorderless"
+        android:clickable="true"
+        app:layout_constraintBottom_toTopOf="@id/shadowLayout"
+        android:focusable="true" />
```
Comment (mukul500) [inline app/src/main/res/layout/fragment_home.xml:64]: Actually I didn't like the idea of an icon on Top. Can u implement Bottom sheet menu and moving favourites icon also on the bottom sheet. Or create a separate fragment for settings.

**Comment by GoluLiliput** [comment]
That should be awesome @mukul500

File: app/src/main/res/layout/exo_player_custom_controls.xml [hunk 4]
```diff
@@ -163,6 +163,22 @@
...
+        android:textColor="@color/media_player_controls"
+        android:text="@string/speed"
+        android:layout_marginEnd="10dp"
+        android:background="?selectableItemBackgroundBorderless"
+        android:textStyle="bold"
+        android:paddingTop="20dp"
+        android:paddingStart="20dp"
+        android:paddingBottom="20dp"
+        android:paddingEnd="10dp"/>
+
```
Comment (mukul500) [inline app/src/main/res/layout/exo_player_custom_controls.xml:181]: This code is also repeated

File: app/src/main/res/layout/fragment_home.xml (hunk 3 above)
Comment (justdvnsh) [reply app/src/main/res/layout/fragment_home.xml:64]: yeah. a separate fragment for settings would be more useful i guess. But what would we be putting in the settings fragment. I can only think of dark mode.

File: app/src/main/res/layout/exo_player_custom_controls.xml (hunk 4 above)
Comment (justdvnsh) [reply app/src/main/res/layout/exo_player_custom_controls.xml:181]: yeah. all the code from my last commit has been repeated. I will surely look at it tomorrow.

**Comment by GoluLiliput** [comment]
Maybe we can add discord invite link for now in the settings panel and write UNDER DEVELOPMENT there until we implement more things there?

**Comment by justdvnsh** [comment]
can be done ! How about it @mukul500

**Comment by mukul500** [comment]
Yeah we need to create a seperate setting or about fragments

**Comment by justdvnsh** [comment]
ok @mukul500 . on it. will be creating a pull request tomorrow.

**Comment by mukul500** [comment]
For Further discussion, u can join Anime Discord #76

**Comment by GoluLiliput** [comment]
Maybe a link to terms and condition inside the setting page???
With the following terms and condition:-



TERMS AND CONDITIONS:-

Anime X Stream only scrapes links from various websites and makes it easier for users to find animes to watch. Anime X Stream or any of it's staffs/developers doesn't host any of the contents found inside Anime X Stream. Any and all images and anime information found on Anime X Stream are taken from it's APIs (Simkl, MyAnimeList and Kitsu) and all of the anime links found on the app are taken from various 3rd Party anime hosting websites. Anime X Stream or it's owners aren't liable for any misuse of any of the contents found inside or outside of the app and cannot be held accountable for the distribution of any of the contents found inside the app. By using Anime X Stream, you accept that the developers of Anime X Stream are not responsible for any of the contents found in the app, nonetheless they be or may not be from their legitimate sources.

**Comment by mukul500** [comment]
1 thing I am gonna make a new branch for this. Please request again for pull request on that branch

**Comment by GoluLiliput** [comment]
Aight... Came up with few stuffs to diversify the setting page of the app... Take a look. I have them noted below.

1).   Default playback quality option. If the available option is not available for an anime or an episode the the quality will automatically set to auto on that specific episode.
Alternative, 
       We can keep these three settings under the playback:- 
     A. Low quality- 360p and 480p
     B. Medium quality- 720p
     C. High quality- 1080p
  *The same automatic auto upon exception case can be implemented in this too.

2).    Separate toogle for automatic theme based on the systemwide dark or light mode. When this option is toggled on, the theme drop-down can be greyed out so that's it becomes inaccessible.

3).    Toggle to switch on/off PIP as the Picture-in-Picture is literally useless in small screen devices which are quite common.

4).    Discord link, privacy policy, terms and condition (which I will provide with the complete .TXT file if approved by u guys), also a GitHub repo button where people are redirect to main page so that they can 'star' and 'watch' the AnimeXStream repository. We aren't very close to the amount of stars that likes of AnYme X has gathered over last many years. N we have just started lol.

5).    Not much of a use but we can add the default playback speed option in settings if required. 

6).    Changelog option which opens up a small page in the app itself which lists the changelogs of all the releases so far. 

I think they are some pretty neat addition to the settings page as of now. Later we can crack our heads more.

@mukul500 @justdvnsh

**Comment by justdvnsh** [comment]
@GoluLiliput The Ideas mentioned seems quite interesting to me. Will be working on it. I have been crazy busy this week, and will be busy till 20th July, due to my internships. However, I would start working on theses further enhancements right after 20th. 
@mukul500

**Comment by GoluLiliput** [comment]
Amazing bro.

**Comment by justdvnsh** [comment]
@GoluLiliput Thanks mate !

**Comment by GoluLiliput** [comment]
Yo bro hw did ur internships go? N u plannin' on the changes yet?

**Comment by justdvnsh** [comment]
@GoluLiliput Internships went amazing. Yes I am going to implement these changes. But, the video player dosen't seem to be working @mukul500 . Whenever we launch any anime to play, the video does not start even on clicking the play button. Seems like some issue. However, I will be working on it. 
Great Man ! I am a 2nd year computer science student. @GoluLiliput where are you from ?

**Comment by GoluLiliput** [comment]
Yeah... Some anime does seem to play at all for me too. The video player doesn't start playing them.

**Comment by GoluLiliput** [comment]
> @GoluLiliput Internships went amazing. Yes I am going to implement these changes. But, the video player dosen't seem to be working @mukul500 . Whenever we launch any anime to play, the video does not start even on clicking the play button. Seems like some issue. However, I will be working on it. 
> Great Man ! I am a 2nd year computer science student. @GoluLiliput where are you from ?

I am from India bro... Madhya Pradesh

**Comment by walkk584** [comment]
Will access to other languages ​​be possible in the sub?

Thanks ;)

**Comment by GoluLiliput** [comment]
> Will access to other languages ​​be possible in the sub?
> 
> Thanks ;)

No

**Comment by ROLYRS** [comment]
Hi 
I have this Version: AnimexStream-v0.2.2-arm64-v8a-releas and I don't see the moon icon to used the Dark UI, I see the video but my app doesn't have that option, can you check that.
I really appreciate it.

BTW
Thanks for this great app, dark will make even better.

**Comment by Pootismancer** [comment]
Hello, I am using "AnimexStream-v0.2.4-universal-release", and I don't see the Dark Mode toggle shown in your demo. Why is that?

**Comment by danielr0206** [comment]
> Hello, I am using "AnimexStream-v0.2.4-universal-release", and I don't see the Dark Mode toggle shown in your demo. Why is that?

Even I'm not getting the manual toggle... can the new version be linked please @mukul500 thanks!

**Pull request closed without merging**